File: backend/src/main/java/com/jbconnect/healthclinix/entity/user/repository/UserModelRepository.java
package com.jbconnect.healthclinix.entity.user.repository;

import com.jbconnect.healthclinix.entity.user.model.UserModel;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import org.springframework.security.core.userdetails.UserDetails;

public interface UserModelRepository extends JpaRepository<UserModel, Long>, JpaSpecificationExecutor<UserModel> {
    UserDetails findByUserName(String userName);
}

File: backendkotlin/src/main/kotlin/com/jbconnect/healthclinix/healthclinix/entity/user/repository/UserModelRepository.kt
package com.jbconnect.healthclinix.healthclinix.entity.user.repository

import com.jbconnect.healthclinix.healthclinix.entity.user.model.UserModel
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor
import org.springframework.security.core.userdetails.UserDetails




interface UserModelRepository: JpaRepository<UserModel, Long>, JpaSpecificationExecutor<UserModel> {

    fun findByUserName(userName: String): UserDetails?

}